### ISSUE TITLE
Problem running with some NPCs

### ISSUE THREAD

**Issue by CrimWolfe** [zeel01/MonsterBlocks#5]
As mentioned, most character sheets are working just fine, but others are giving me an error. This is what had shown up.

foundry.js:3833 TypeError: An error occurred while rendering MonsterBlock5e 70: Cannot read property 'target' of undefined
    at Function.isLegendaryResistance (monsterblock.js:392)
    at MonsterBlock5e.getLegendaryResistance (monsterblock.js:145)
    at MonsterBlock5e.getData (monsterblock.js:42)
    at MonsterBlock5e._render (foundry.js:3861)
    at MonsterBlock5e._render (foundry.js:4473)
    at MonsterBlock5e._render (foundry.js:17411)
    at MonsterBlock5e.render (foundry.js:3831)
    at ActorDirectory._onClickEntityName (foundry.js:39639)
    at HTMLOListElement.dispatch (jquery.min.js:2)
    at HTMLOListElement.v.handle (jquery.min.js:2)

**Comment by zeel01** [comment]
Is the creature in question manually entered, homebrew, SRD, or imported with a module?

**Comment by CrimWolfe** [comment]
Foundry VTT | Importing Actor Sprite from SharedData.monsters
foundry.js:3855 Foundry VTT | Rendering ActorSheet5eNPCDark
foundry.js:9682 Foundry VTT | Created Actor with id W6G1p7b2XJSPmRnO
foundry.js:3855 Foundry VTT | Rendering ActorSheet5eNPCDark
foundry.js:3855 Foundry VTT | Rendering EntitySheetConfig
foundry.js:3855 Foundry VTT | Rendering MonsterBlock5e
foundry.js:5224 Foundry VTT | Retrieved and compiled template modules/monsterblock/actor-sheet.html
foundry.js:3833 TypeError: An error occurred while rendering MonsterBlock5e 92: Cannot read property '0' of undefined
    at Object.damageformula (monsterblock.js:562)
    at Object.c (handlebars.min.js:27)
    at eval (eval at createFunctionContext (handlebars.min.js:29), <anonymous>:40:159)
    at Object.h [as fn] (handlebars.min.js:27)
    at Object.<anonymous> (handlebars.min.js:27)
    at Object.c (handlebars.min.js:27)
    at eval (eval at createFunctionContext (handlebars.min.js:29), <anonymous>:15:47)
    at Object.h [as fn] (handlebars.min.js:27)
    at Object.<anonymous> (handlebars.min.js:27)
    at Object.c (handlebars.min.js:27)

**Comment by CrimWolfe** [comment]
> Is the creature in question manually entered, homebrew, SRD, or imported with a module?

I tried from SRD and the imported one with VTTA. Both doing the same thing

**Comment by CrimWolfe** [comment]
and now the SRD one is working and the VTTA one isn't ...

**Comment by zeel01** [comment]
Could you also confirm that this happens when Monster Blocks is the only module in use?

**Comment by zeel01** [comment]
Possible additional example from another Discord user:
```
@zeel 

TypeError: An error occurred while rendering MonsterBlock5e 34: Cannot read property '0' of undefined
    at Object.damageformula (monsterblock.js:568)
    at Object.c (handlebars.min.js:27)
    at eval (eval at createFunctionContext (handlebars.min.js:29), <anonymous>:40:159)
    at Object.h [as fn] (handlebars.min.js:27)
    at Object.<anonymous> (handlebars.min.js:27)
    at Object.c (handlebars.min.js:27)
    at eval (eval at createFunctionContext (handlebars.min.js:29), <anonymous>:15:47)
    at Object.h [as fn] (handlebars.min.js:27)
    at Object.<anonymous> (handlebars.min.js:27)
    at Object.c (handlebars.min.js:27)
(anonymous) @ foundry.js:3833
```
Using SRD Giant Spider.

**Comment by Jonwh25** [comment]
Issue with trying to load Nezzner the Black Spider which I imported from dndbeyond

foundry.js:3833 SyntaxError: An error occurred while rendering MonsterBlock5e 73: Invalid or unexpected token
    at new Function (<anonymous>)
    at Roll._safeEval (foundry.js:7681)
    at Roll.roll (foundry.js:7520)
    at Function.maximize (foundry.js:7899)
    at Object.averagedamage (monsterblock.js:550)
    at Object.c (handlebars.min.js:27)
    at eval (eval at createFunctionContext (handlebars.min.js:29), <anonymous>:44:159)
    at Object.h [as fn] (handlebars.min.js:27)
    at Object.<anonymous> (handlebars.min.js:27)
    at Object.c (handlebars.min.js:27)

Here's one for a spectator which I also imported from dndbeyond

foundry.js:3833 SyntaxError: An error occurred while rendering MonsterBlock5e 70: Invalid or unexpected token
    at new Function (<anonymous>)
    at Roll._safeEval (foundry.js:7681)
    at Roll.roll (foundry.js:7520)
    at Function.maximize (foundry.js:7899)
    at Object.averagedamage (monsterblock.js:550)
    at Object.c (handlebars.min.js:27)
    at eval (eval at createFunctionContext (handlebars.min.js:29), <anonymous>:44:159)
    at Object.h [as fn] (handlebars.min.js:27)
    at Object.<anonymous> (handlebars.min.js:27)
    at Object.c (handlebars.min.js:27)

This pops up randomly when going between different monsters

monsterblock.js:118 Uncaught (in promise) TypeError: Cannot read property 'classList' of undefined
    at MonsterBlock5e.setCurrentTheme (monsterblock.js:118)
    at monsterblock.js:11

**Comment by CrimWolfe** [comment]
> Could you also confirm that this happens when Monster Blocks is the only module in use?

Confirmed, still happens. I still have some that even putting them back, here's a Tressym (Monstrosity) imported from dndbeyond..

foundry.js:3833 TypeError: An error occurred while rendering MonsterBlock5e 39: Cannot read property '0' of undefined
    at Object.damageformula (monsterblock.js:568)
    at Object.c (handlebars.min.js:27)
    at eval (eval at createFunctionContext (handlebars.min.js:29), <anonymous>:40:159)
    at Object.h [as fn] (handlebars.min.js:27)
    at Object.<anonymous> (handlebars.min.js:27)
    at Object.c (handlebars.min.js:27)
    at eval (eval at createFunctionContext (handlebars.min.js:29), <anonymous>:15:47)
    at Object.h [as fn] (handlebars.min.js:27)
    at Object.<anonymous> (handlebars.min.js:27)
    at Object.c (handlebars.min.js:27)

**Comment by nstiurca** [comment]
This also happens for me. In particular, I tried it with a Goblin.

Foundry v0.6.5
DnD 5e v0.93
Monster Blocks v1.1.1
(no other modules)
Windows 10 native app

It also happens when running the server in headless mode on an Ubuntu machine and connecting to it from a Chrome browser on a Windows machine. Ie, it does not seem to be OS-dependent nor client-dependent as far as I can tell.

Detailed log:

```
foundry.js:3833 TypeError: An error occurred while rendering MonsterBlock5e 49: Cannot read property '0' of undefined
    at Object.damageformula (monsterblock.js:568)
    at Object.c (handlebars.min.js:27)
    at eval (eval at createFunctionContext (handlebars.min.js:29), <anonymous>:40:159)
    at Object.h [as fn] (handlebars.min.js:27)
    at Object.<anonymous> (handlebars.min.js:27)
    at Object.c (handlebars.min.js:27)
    at eval (eval at createFunctionContext (handlebars.min.js:29), <anonymous>:15:47)
    at Object.h [as fn] (handlebars.min.js:27)
    at Object.<anonymous> (handlebars.min.js:27)
    at Object.c (handlebars.min.js:27)
-- | -- | -- | --
  | (anonymous) | @ | foundry.js:3833
  | Promise.catch (async) |   |  
  | render | @ | foundry.js:3831
  | render | @ | foundry.js:9281
  | _onUpdate | @ | foundry.js:26190
  | _onUpdateTokenActor | @ | foundry.js:38355
  | _onUpdate | @ | foundry.js:38258
  | _onUpdateEmbeddedEntity | @ | foundry.js:29070
  | _handleUpdateEmbeddedEntity | @ | foundry.js:10204
  | _handleUpdateEmbeddedEntity | @ | foundry.js:29118
  | updateEmbeddedEntity | @ | foundry.js:10173
  | async function (async) |   |  
  | updateEmbeddedEntity | @ | foundry.js:10163
  | update | @ | foundry.js:11938
  | update | @ | foundry.js:26341
  | update | @ | foundry.js:26131
  | update | @ | entity.js:350
  | prepFlags | @ | monsterblock.js:217
  | MonsterBlock5e | @ | monsterblock.js:9
  | get sheet | @ | foundry.js:9376
  | _onUpdateTokenActor | @ | foundry.js:38367
  | _onUpdate | @ | foundry.js:38258
  | _onUpdateEmbeddedEntity | @ | foundry.js:29070
  | _handleUpdateEmbeddedEntity | @ | foundry.js:10204
  | _handleUpdateEmbeddedEntity | @ | foundry.js:29118
  | updateEmbeddedEntity | @ | foundry.js:10173
  | async function (async) |   |  
  | updateEmbeddedEntity | @ | foundry.js:10163
  | update | @ | foundry.js:11938
  | update | @ | foundry.js:26341
  | update | @ | foundry.js:26131
  | update | @ | entity.js:350
  | setFlag | @ | foundry.js:10402
  | (anonymous) | @ | monsterblock.js:916
  | async function (async) |   |  
  | (anonymous) | @ | monsterblock.js:915
```

**Comment by zeel01** [comment]
@Jonwh25 That looks like it's a different problem than the others, could you please post a separate issue and include an export of the monsters in question.

**Comment by zeel01** [comment]
The main issue seems to be that many creatures from the SRD or imported from elsewhere have incomplete data. For the most part, this shouldn't cause errors anymore. Though you might see the word "undefined" pop up in some places on the sheet.

Most if not all of these errors should be solved in v1.2.0
